core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -186,6 +292,12 @@ public T get(Settings settings) {
      * instead. This is useful if the value can't be parsed due to an invalid value to access the actual value.
      */
     public String getRaw(Settings settings) {
+        // They're using the setting, so we need to tell them to stop
+        if (this.isDeprecated() && this.exists(settings)) {
+            // It would be convenient to show its replacement key, but replacement is often not so simple
+            deprecationLogger.deprecated("[{}] setting was deprecated in Elasticsearch and it will be removed in a future release! " +
+                    "See the breaking changes lists in the documentation for details", getKey());
```

Comment:
Incredibly minor: should match the value used by the `return` line so that if the method ever changes, we use the same value.